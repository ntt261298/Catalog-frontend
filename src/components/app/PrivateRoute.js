import React from 'react';
import { BrowserRouter as Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

export const PrivateRoute = ({ component: Component, token, ...rest }) => (
  <Route
    {...rest}
    render={props => (
      !token
        ? <Redirect to="/" />
        : <Component {...props} />
    )
    }
  />
);

PrivateRoute.propTypes = {
  component: PropTypes.func,
  token: PropTypes.string,
};

PrivateRoute.defaultProps = {
  token: '',
  component() {},
};

export default PrivateRoute;
