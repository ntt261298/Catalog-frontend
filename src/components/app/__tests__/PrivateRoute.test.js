import React from 'react';
import { shallow } from '../../../enzyme';
import { PrivateRoute } from '../PrivateRoute';

describe('src/components/home/HomePage.js', () => {
  it('should render private route', () => {
    const props = {
      components: '',
      token: 'randomToken',
    };
    const wrapper = shallow(<PrivateRoute {...props} />);
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
  it('should not render private route', () => {
    const props = {
      components: '',
      token: '',
    };
    const wrapper = shallow(<PrivateRoute {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
