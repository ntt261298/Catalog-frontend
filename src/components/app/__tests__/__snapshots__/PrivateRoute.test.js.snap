// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/home/HomePage.js should not render private route 1`] = `
<BrowserRouter
  components=""
  render={[Function]}
/>
`;

exports[`src/components/home/HomePage.js should render private route 1`] = `
<BrowserRouter
  components=""
  render={[Function]}
/>
`;
